### ISSUE TITLE
Hololens2 Unity Windows XR Plugin support

### ISSUE THREAD

**Issue by thomaswester** [Azure/azure-spatial-anchors-samples#212]
The current samples use Unity built-in VR support, however this is being deprecated and Windows XR Plugin is taking it's place. 

Currently my hololens2 app crashes when I set CloudSpatialAnchor.LocalAnchor to WorldAnchor.GetNativeSpatialAnchorPtr. The app is configured to use Windows XR Plugin 2.3, AR Foundation 2.1.8 and AR Subsystem 2.1.2.

WorldAnchor is in this Unity.XR.WSA namespace, which is part of the built-in VR support that Unity is deprecating.

My code is based on this tutorial:
https://docs.microsoft.com/en-us/azure/spatial-anchors/tutorials/tutorial-new-unity-hololens-app?tabs=UnityPackage

**Comment by Aupuma** [comment]
Same issue here. I'm trying to follow this tutorial:
https://docs.microsoft.com/es-es/windows/mixed-reality/mr-learning-asa-02

The session starts fine, but when I try to create an anchor it crashes to WorldAnchor.GetNativeSpatialAnchorPtr.

**Comment by kaysabelle** [comment]
Hi @thomaswester, thanks for reporting this. We will investigate it and get back to you (internal issue #29379850).

**Comment by thomaswester** [comment]
@kaysabelle I just found "World Locking Tools". This looks like the way forward, beyond WorldAnchor. Can you confirm?
https://microsoft.github.io/MixedReality-WorldLockingTools-Unity/README.html

**Comment by kaysabelle** [comment]
Hi @thomaswester, at the moment Azure Spatial Anchors currently supports Unity 2019.4 (LTS). There is a deprecation warning for Unity's built-in VR support, but this is what the SDK uses today to support Unity 2019.4.

We are working on support for Unity 2020 as well as the new XR Plugin system in future versions of the Azure Spatial Anchors SDK.

**Comment by raymondpols** [comment]
Hi, any progress on this one yet? I’m experiencing the same issue: spatial anchor creation worked fine on Unity 2019.4 and HoloLens1. Now with Unity 2020 and HoloLens 2 (and new XR system) creating anchors on HoloLens 2 crashes. Same code on iOS works though!

**Comment by msftradford** [comment]
Hi @raymondpols, thanks for reaching out! ASA is available on HoloLens 2, but we only support Unity 2019.4. Thanks!

**Comment by raymondpols** [comment]
Thanks for the quick feedback. When do you think Unity 2020 is supported? Also, when we test ASA (using same Unity 2020 project) on iOS then it does work. Any explanation for that?

**Comment by msftradford** [comment]
You can learn a little bit more about what is going on behind the scenes [here](https://github.com/Azure/azure-spatial-anchors-samples/issues/219). Because of this, we cannot currently commit to a timeline for Unity 2020 support.

**Comment by raymondpols** [comment]
Alright. Thanks. So it’s effect the new Unity XR system that causes the issue. So not only downgrade to Unity 2019, but also go back to the old VR system?

**Comment by msftradford** [comment]
That is correct. We'll be sure to update the release notes once we're ready for Unity 2020.

**Comment by varunsiddaraju** [comment]
I'm working on ASA lesson for HoloLens 2: https://docs.microsoft.com/es-es/windows/mixed-reality/mr-learning-asa-02 

I'm using Unity 2019.4 LTS and Unity XR system. I'm able to successfully build and deploy the app to HoloLens 2 but my app crashes whenever I call CreateAnchor( ) function.

Could someone please confirm me whether Azure spatial anchor SDK works for with Unity 2019.4 XR system?

If yes, please let me know if you know the solution for the above problem.

Thanks :)

**Comment by msftradford** [comment]
Hi @varunsiddaraju, thanks for the question. Yes, the ASA SDK does work with Unity 2019.4. Unfortunately, the page that you reference seems a bit out of date at the moment. A [ticket](https://github.com/MicrosoftDocs/mixed-reality/issues/142) has been opened to address that issue. If you want to create a Unity/HoloLens project, please take a look at the ASA documentation [quickstart page](https://docs.microsoft.com/en-us/azure/spatial-anchors/quickstarts/get-started-unity-hololens?tabs=azure-portal). If you are having an issue after following the quickstart, please file a new issue. Thanks!

**Comment by israaz** [comment]
Hello, 
I am facing the same issue:
I am using Unity 2021.3.16f1 LTS and Unity XR system.
I am working on this tutorial:
https://learn.microsoft.com/en-us/training/modules/azure-spatial-anchors-tutorials/3-exercise-get-started-with-azure-spatial-anchors

I followed the tutorial step by step and built and deployed the application on Hololens 2 and no configuration errors. However, when I started the app, I followed the following instructions as in the tutorial:

When the app runs on your device, follow the on-screen instructions displayed on the Azure Spatial Anchor Tutorial Instructions panel:
Move the cube to a different location.
Start an Azure session.
Create an Azure anchor at the location of the cube.
Stop the Azure session.
Remove the local anchor to allow the user to move the cube.
Move the cube to somewhere else.
Start an Azure session.
Find the Azure anchor to position the cube at the location from step 3.
Delete the Azure anchor.
Stop the Azure session.


In step 3 (create azure anchor), the application closes directly. So, is this related to the spatial anchors SDK? Does Azure spatial anchor SDK not work with the Unity 2021 XR system?
